frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts

```diff
@@ -110,30 +122,22 @@ export const prepareRecordingSnapshots = (
     newSnapshots?: RecordingSnapshot[],
     existingSnapshots?: RecordingSnapshot[]
 ): RecordingSnapshot[] => {
-    const seenHashes: Record<string, (RecordingSnapshot | string)[]> = {}
+    const seenHashes: Set<string> = new Set()
 
     return (newSnapshots || [])
         .concat(existingSnapshots ? existingSnapshots ?? [] : [])
         .filter((snapshot) => {
             // For a multitude of reasons, there can be duplicate snapshots in the same recording.
-            // We can deduplicate by filtering out snapshots with the same timestamp and delay value (this is quite unique as a pairing)
-            const key = `${snapshot.timestamp}-${snapshot.delay}`
+            // we have to stringify the snapshot to compare it to other snapshots.
+            // so we can filter by storing them all in a set
```

Comment:
I wonder if this will affect the performance of the browser considerably, not sure how big snapshots are and how many snapshots a replay usually has but I imagine this can be an issue for low-end devices, have you checked it?

Reply:
we're already `JSON.stringify`ing the snapshot in that scope so I don't think we're holding too much more. 

we could hash the string but I think if we're causing a problem here we're already causing a problem here :)

Reply:
I see, I somehow thought that only the `const key = ${snapshot.timestamp}-${snapshot.delay}` was stored in memory, but you're right, my bad, it was a`Record<string, (RecordingSnapshot | string)[]>` already hence the problem already exists anyway, likely not a biggie then, go go go.